src/gui/mod.rs

```diff
@@ -0,0 +1,115 @@
+pub mod helpers;
+pub struct Gui;
+use super::state::State;
+use eframe::egui;
+use std::time::Instant;
+use egui_dock::{DockArea, Style};
+use super::gui::helpers::GuiHelper;
+use super::memory::MemoryManagers;
+
+pub struct TabViewer<'a> {
+    helpers: &'a Vec<Box<dyn GuiHelper>>,
+    memory_managers: &'a MemoryManagers,
+}
+impl TabViewer<'_> {
+    fn draw(&mut self, ui: &mut egui::Ui, tab: &mut String, name: String) {
+        if let Some(helper) = self.helpers.iter().find(|x| {
+            x.name() == name
+        }) {
+            helper.draw(self.memory_managers, ui, tab);
+        }
+    }
+}
+
+impl egui_dock::TabViewer for TabViewer<'_> {
+    type Tab = String;
+
+    fn title(&mut self, tab: &mut Self::Tab) -> egui::WidgetText {
+        (&*tab).into()
+    }
+
+    fn ui(&mut self, ui: &mut egui::Ui, tab: &mut Self::Tab) {
+        self.draw(ui, tab, tab.to_string());
+    }
+}
+
+impl Gui {
+    pub fn run() {
+        let options = eframe::NativeOptions {
+            viewport: egui::ViewportBuilder::default()
+                .with_inner_size([800.0, 800.0])
+                .with_min_inner_size([800.0, 800.0]),
+
+            ..Default::default()
+        };
+        eframe::run_native(
+            "Sea of Stars TAS",
+            options,
+            Box::new(|cc| Ok(Box::new(State::new(cc)))),
```

Comment:
Not gonna lie, I'm not sure what's going on here. Too many boxes :P
I assume at least that this is where the `State` is created, which in turn starts initializing memory stuff?